Quake/sys_sdl_unix.c

```diff
@@ -240,138 +241,98 @@ static int Sys_NumCPUs (void)
 #include <FindDirectory.h>
 #include <fs_info.h>
-
 static void Sys_GetUserdir (char *dst, size_t dstsize)
 {
 	dev_t volume = dev_for_path("/boot");
 	char buffer[B_PATH_NAME_LENGTH];
 	status_t result;
 
-	result = find_directory(B_USER_NONPACKAGED_DATA_DIRECTORY, volume, false, buffer, sizeof(buffer));
+	result = find_directory(B_USER_SETTINGS_DIRECTORY, volume, false, buffer, sizeof(buffer));
 	if (result != B_OK)
 		Sys_Error ("Couldn't determine userspace directory");
 
 	q_snprintf (dst, dstsize, "%s/%s", buffer, SYS_USERDIR);
 }
-#else
-static void Sys_GetUserdir (char *dst, size_t dstsize)
+static void Sys_GetBasedir (char *dst, size_t dstsize)
 {
-	size_t		n;
-	const char	*home_dir = NULL;
-	struct passwd	*pwent;
+	dev_t volume = dev_for_path("/boot");
+	char buffer[B_PATH_NAME_LENGTH];
+	status_t result;
 
-	pwent = getpwuid( getuid() );
-	if (pwent == NULL)
-		perror("getpwuid");
-	else
-		home_dir = pwent->pw_dir;
-	if (home_dir == NULL)
-		home_dir = getenv("HOME");
-	if (home_dir == NULL)
+	result = find_directory(B_USER_NONPACKAGED_DATA_DIRECTORY, volume, false, buffer, sizeof(buffer));
+	if (result != B_OK)
 		Sys_Error ("Couldn't determine userspace directory");
 
-/* what would be a maximum path for a file in the user's directory...
- * $HOME/SYS_USERDIR/game_dir/dirname1/dirname2/dirname3/filename.ext
- * still fits in the MAX_OSPATH == 256 definition, but just in case :
- */
-	n = strlen(home_dir) + strlen(SYS_USERDIR) + 50;
-	if (n >= dstsize)
-		Sys_Error ("Insufficient array size for userspace directory");
-
-	q_snprintf (dst, dstsize, "%s/%s", home_dir, SYS_USERDIR);
-}
-#endif	/* PLATFORM_HAIKU */
-#endif	/* DO_USERDIRS */
-
-#ifdef PLATFORM_OSX
-static char *OSX_StripAppBundle (char *dir)
-{ /* based on the ioquake3 project at icculus.org. */
-	static char	osx_path[MAX_OSPATH];
-
-	q_strlcpy (osx_path, dir, sizeof(osx_path));
-	if (strcmp(basename(osx_path), "MacOS"))
-		return dir;
-	q_strlcpy (osx_path, dirname(osx_path), sizeof(osx_path));
-	if (strcmp(basename(osx_path), "Contents"))
-		return dir;
-	q_strlcpy (osx_path, dirname(osx_path), sizeof(osx_path));
-	if (!strstr(basename(osx_path), ".app"))
-		return dir;
-	q_strlcpy (osx_path, dirname(osx_path), sizeof(osx_path));
-	return osx_path;
+	q_snprintf (dst, dstsize, "%s/%s", buffer, SYS_USERDIR);
 }
-
-static void Sys_GetBasedir (char *argv0, char *dst, size_t dstsize)
+#else /* PLATFORM_HAIKU */
+#define SYS_USERDIR "quakespasm"
+static void Sys_GetUserdir (char *dst, size_t dstsize)
 {
-	char	*tmp;
-
-	if (realpath(argv0, dst) == NULL)
+#ifdef USE_SDL2
+	char *prefpath = SDL_GetPrefPath (NULL, SYS_USERDIR);
+	if (prefpath == NULL)
+		Sys_Error ("Couldn't determine userspace directory");
+	q_snprintf (dst, dstsize, "%s", prefpath);
+	SDL_free (prefpath);
+#else /* USE_SDL2 */
+	/* note - we're specifically not using XDG_CONFIG_HOME here so we can match
+	 * the behaviour of SDL_GetPrefPath()
+	 */
+	const char *env = getenv("XDG_DATA_HOME");
```

Comment:
I like that XDG paths are being employed, one less dot file/folder in $HOME.  But should there be a fallback to $HOME/.quakespasm so as to not break current installations?

Reply:
Maybe it could check if there's an existing `$HOME/.quakespasm` folder and use that if so?

Reply:
That sounds reasonable